### PR TITLE
ADBDEV-5263: Fix incorrect file descriptor dup when creating a pipe

### PR DESCRIPTION
**Pull request by KnightMurloc**
Fix incorrect file descriptor dup when creating a pipe

If fcntl fails with F_DUPFD_CLOEXEC, attempts are made to duplicate the file
descriptor using a regular dup. But since, as a result of fcntl failure,
write_fd will become -1 and dup will also fail.

This patch returns write_fd to its original value in case of fcntl failure.

### CODE REVIEW

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/67992

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1228622

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1228623

**Review by RekGRpth — DISMISSED** [review]
put title in PR description

File: gpMgmt/bin/pythonSrc/subprocess32/_posixsubprocess.c [hunk 1]
```diff
@@ -791,6 +791,7 @@ subprocess_cloexec_pipe(PyObject *self, PyObject *noargs)
         if (write_fd < 0)  /* We don't support F_DUPFD_CLOEXEC / other error */
 #endif
         {
+            write_fd = fds[1];
```
Comment (andr-sokolov) [inline gpMgmt/bin/pythonSrc/subprocess32/_posixsubprocess.c:795]: When F_DUPFD_CLOEXEC is not defined, `write_fd = fds[1]` is executed twice: here and at the 785th line.

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/68096

**Review by RekGRpth — APPROVED** [review]

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1230767

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1230768

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/68125

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1233776

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1233777

**Pull request merged**